provider/equinix/environ.go

```diff
@@ -302,20 +307,38 @@ func (e *environ) StartInstance(ctx context.ProviderCallContext, args environs.S
 		return nil, errors.Trace(err)
 	}
 
-	if err := e.finishInstanceConfig(&args, spec); err != nil {
-		return nil, errors.Trace(err)
-	}
+	return spec, nil
+}
 
+func getCloudConfig(args environs.StartInstanceParams) (cloudinit.CloudConfig, error) {
 	cloudCfg, err := cloudinit.New(args.InstanceConfig.Series)
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	cloudCfg.AddPackage("iptables-persistent")
+
+	// Set a default INPUT policy of drop, permitting ssh
+	iptablesDefault := []string{
+		"iptables -A INPUT -m conntrack --ctstate ESTABLISHED,RELATED -j ACCEPT",
+		"iptables -A INPUT -m state --state ESTABLISHED,RELATED -j ACCEPT",
+		"iptables -A INPUT -p tcp --dport ssh -j ACCEPT",
```

Comment:
My rules may be insufficient:
```
Setting up dns-root-data (2019052802) ...
Setting up libidn11:amd64 (1.33-2.2ubuntu2) ...
Setting up bridge-utils (1.6-2ubuntu1) ...
Setting up dnsmasq-base (2.80-1.1ubuntu1.5) ...
Setting up ubuntu-fan (0.12.13ubuntu0.1) ...
Created symlink /etc/systemd/system/multi-user.target.wants/ubuntu-fan.service → /lib/systemd/system/ubuntu-fan.service.
Processing triggers for systemd (245.4-4ubuntu3.18) ...
Processing triggers for man-db (2.9.1-1) ...
Processing triggers for dbus (1.12.16-2ubuntu2.2) ...
Processing triggers for libc-bin (2.31-0ubuntu9.9) ...
728a5565b2f05e251b14f9f44a695e05f1d38ecc79830ce28d81ce8c99a76939  /var/lib/juju/tools/2.9.35.1-ubuntu-amd64/tools.tar.gz
925d99bcc83bb9accc88df729d12e5c83a07ece73a94f936fbbae959184ac12d  /var/lib/juju/gui/gui.tar.bz2
Installing Juju machine agent
2022-10-11 17:06:10 INFO juju.cmd supercommand.go:56 running jujud [2.9.35.1  gc go1.18.4]
2022-10-11 17:06:11 INFO juju.agent identity.go:22 writing system identity file
2022-10-11 17:06:11 WARNING juju.mongo mongo.go:481 overwriting args.dataDir (set to /var/lib/juju) to /var/snap/juju-db/common
2022-10-11 17:06:11 INFO juju.mongo mongo.go:486 Ensuring mongo server is running; data directory /var/snap/juju-db/common; port 37017
2022-10-11 17:06:11 WARNING juju.mongo service.go:323 configuring mongod  with --noauth flag enabled
2022-10-11 17:06:11 INFO juju.packaging manager.go:103 installing "juju-db" via "snap"
2022-10-11 17:06:11 INFO juju.packaging.manager run.go:88 Running: snap install  --channel 4.4/stable juju-db
2022-10-11 17:12:55 INFO juju.worker.peergrouper initiate.go:68 finished InitiateMongoServer
ERROR failed to start mongo: cannot initiate replica set: cannot dial mongo to initiate replicaset: no reachable servers
ERROR failed to bootstrap model: subprocess encountered error code 1
```

Reply:
I believe for controllers you will need to expose the Mongo port, which can be accessed via:
```go
args.InstanceConfig.ControllerConfig.StatePort()
```

Reply:
The default ruleset is now exposing the following for Controllers:			
```
args.InstanceConfig.ControllerConfig.APIPort(),
args.InstanceConfig.ControllerConfig.StatePort(),
args.InstanceConfig.ControllerConfig.ControllerAPIPort(),
```